geary-papermc-tracking/src/main/kotlin/com/mineinabyss/geary/papermc/tracking/blocks/BlockTracking.kt

```diff
@@ -0,0 +1,95 @@
+package com.mineinabyss.geary.papermc.tracking.blocks
+
+import com.mineinabyss.geary.addons.dsl.GearyAddonWithDefault
+import com.mineinabyss.geary.modules.geary
+import com.mineinabyss.geary.papermc.tracking.blocks.helpers.GearyBlockPrefabQuery
+import com.mineinabyss.geary.papermc.tracking.blocks.systems.TrackOnSetBlockComponent
+import com.mineinabyss.geary.prefabs.PrefabKey
+import com.mineinabyss.idofront.di.DI
+import org.bukkit.Instrument
+import org.bukkit.Material
+import org.bukkit.Note
+import org.bukkit.block.BlockFace
+import org.bukkit.block.data.BlockData
+import org.bukkit.block.data.type.CaveVines
+import org.bukkit.block.data.type.NoteBlock
+import org.bukkit.block.data.type.Tripwire
+
+val gearyBlocks by DI.observe<BlockTracking>()
+interface BlockTracking {
+    val blockMap: Map<BlockData, Int>
+    val blockProvider: GearyBlockProvider
+    val block2Prefab: Block2Prefab
+    val blockPrefabs: GearyBlockPrefabQuery
+
+    fun createBlockData(prefabKey: PrefabKey): BlockData? = blockProvider.serializePrefabToBlockData(prefabKey)
+
+    companion object : GearyAddonWithDefault<BlockTracking> {
+        override fun default(): BlockTracking = object : BlockTracking {
+            override val blockMap = createBlockMap()
+            override val blockProvider = GearyBlockProvider()
+            override val blockPrefabs = GearyBlockPrefabQuery()
+            override val block2Prefab = Block2Prefab()
+        }
+
+        override fun BlockTracking.install() {
+            geary.pipeline.addSystems(
+                TrackOnSetBlockComponent()
+            )
+        }
+
+        private fun createBlockMap(): MutableMap<BlockData, Int> {
+            return mutableMapOf<BlockData, Int>().apply {
+                // Calculates tripwire states
+                for (i in 0..127) {
+                    val tripWireData = Material.TRIPWIRE.createBlockData() as Tripwire
+                    if (i and 1 == 1) tripWireData.setFace(BlockFace.NORTH, true)
+                    if (i shr 1 and 1 == 1) tripWireData.setFace(BlockFace.EAST, true)
+                    if (i shr 2 and 1 == 1) tripWireData.setFace(BlockFace.SOUTH, true)
+                    if (i shr 3 and 1 == 1) tripWireData.setFace(BlockFace.WEST, true)
+                    if (i shr 4 and 1 == 1) tripWireData.isPowered = true
+                    if (i shr 5 and 1 == 1) tripWireData.isDisarmed = true
+                    if (i shr 6 and 1 == 1) tripWireData.isAttached = true
+
+                    putIfAbsent(tripWireData, i)
+                }
```

Comment:
Calculates the TripWire BlockDatas. 128 total states, reserves 1 state (0) as we want 1 normal string/tripwire